tests/runtimes/android/android_app_gradle_tests.py

```diff
@@ -0,0 +1,235 @@
+         https://github.com/NativeScript/android-runtime/issues/993
+        """
+
+        # Change main-page.js so it contains only logging information
+        source_js = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-993', "api",
+                                 'main-page.js')
+        target_js = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'main-page.js')
+        File.copy(src=source_js, target=target_js)
+        # Change app app.gradle so it contains the dependencies to com.github.myinnos:AwesomeImagePicker:1.0.2
+        source_js = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-993', "api",
+                                 'app.gradle')
+        target_js = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'App_Resources', 'Android', 'app.gradle')
+        File.copy(src=source_js, target=target_js)
+        Tns.platform_remove(app_name=APP_NAME, platform=Platform.ANDROID)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        log = Tns.run_android(APP_NAME, device=self.emulator.id, wait=False, verify=False)
+
+        strings = ['Project successfully built',
+                   'Successfully installed on device with identifier', self.emulator.id,
+                   'Successfully synced application', '###TEST API PASSED###']
+
+        test_result = Wait.until(lambda: all(string in File.read(log.log_file) for string in strings), timeout=300,
+                                 period=5)
+        assert test_result, 'Native packages could not be used with api in app.gradle'
+
+    def test_319_build_project_with_foursquare_android_oauth(self):
+        # This is required when build with different SDK
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(APP_NAME, template=Template.HELLO_WORLD_JS.local_package)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+
+        # Add foursquare native library as dependency
+        source = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-755',
+                              'app.gradle')
+        target = os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'App_Resources', 'Android', 'app.gradle')
+        File.copy(src=source, target=target)
+
+        # Build the project
+        output = Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME), verify=False)
+        assert ':asbg:generateBindings', 'Static Binding Generator not executed'
+        assert 'cannot access its superclass' not in output.output
+
+    def test_420_include_gradle_flavor(self):
+        # https://github.com/NativeScript/android-runtime/pull/937
+        # https://github.com/NativeScript/nativescript-cli/pull/3467
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
```

Comment:
Can we use Template.MIN_JS?

Reply:
All tests are based on hello world js template. I would prefer using it.